src/Microsoft.AspNetCore.Server.Kestrel/Http/DateHeaderValueManager.cs

```diff
@@ -65,7 +64,7 @@ public DateHeaderValues GetDateHeaderValues()
 
             return new DateHeaderValues()
             {
-                Bytes = _activeDateBytes ? _dateBytes0 : _dateBytes1,
+                Bytes = _dateBytes,
```

Comment:
It seems like it's possible (though very unlikely) that you could get mismatching values for the bytes and the string, no?

Why don't we make DateHeaderValues a class to deal with this?  Slightly less work in GetDateHeaderValues too.